File: Novels/packages/6-mobile-android/StoryReader/app/src/main/java/com/storysphere/storyreader/tts/TTSEngine.kt
package com.storysphere.storyreader.tts

interface TTSEngine {
    suspend fun speak(text: String, onUtteranceCompleted: (() -> Unit)? = null)
    suspend fun stop()
    suspend fun pause()
    suspend fun resume()
    fun setSpeechRate(rate: Float) // 0.5x to 2.0x
    fun setPitch(pitch: Float) // 0.5 to 2.0
    fun isSpeaking(): Boolean
    fun shutdown()
}

File: Novels/packages/5-mobile-ios/StoryReader/Service/TTS/TTSEngine.swift
import Foundation
import AVFoundation

// TTS Engine Protocol
protocol TTSEngine {
    func synthesize(text: String, voice: String?, speed: Double) async throws -> Data
    func speak(text: String, voice: String?, speed: Double) async throws
    func stop()
    func pause()
    func resume()
    var isSpeaking: Bool { get }
    var currentProgress: Double { get } // 0.0 - 1.0
}

// Shared TTS Error enum
enum TTSError: Error {
    case engineNotAvailable
    case synthesisFailed
    case invalidVoice
    case notSupported
}

